packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -0,0 +1,562 @@
+  /// Selection does not reach this selectable and should look previous
+  /// selectable in screen order.
+  previous,
+  /// Selection ends in this selectable.
+  ///
+  /// part of the selectable may or may not be selected, but there are still
+  /// content to select forward or backward.
+  end,
+  /// The result can't be determine in this frame.
+  ///
+  /// This is typically used when the subtree is scrolling to reveal more content.
+  pending,
+  /// There is no result for the selection event.
+  ///
+  /// This is used when the a selection result is not applicable, e.g.
+  /// [SelectAllSelectionEvent], [ClearSelectionEvent], and
+  /// [SelectWordSelectionEvent].
+  none,
+}
+
+/// The abstract interface to handle selection events.
+///
+/// {@template flutter.rendering.SelectionHandler}
+/// This class returns a [SelectionGeometry] as its [value], and is responsible
+/// to notify its listener when its selection geometry has changed as the result
+/// of receiving selection events.
+/// {@endtemplate}
+abstract class SelectionHandler implements ValueListenable<SelectionGeometry?> {
+  /// Marks this handler to be responsible for painting leader layers for the
+  /// selection handles.
+  ///
+  /// This handler is responsible for painting the leader layers with the
+  /// given layer links if they are not null. It is possible only one layer is
+  /// non-null if this handler is only responsible for painting that non-null
+  /// layer link.
+  ///
+  /// If you are implementing this method as a result of mixing [Selectable] to
+  /// a leaf render object, consider also mixing the [HandleLayersPainterMixin]
+  /// to get the default implementation.
+  ///
+  /// See also:
+  ///  * [HandleLayersPainterMixin]: which paint layer links automatically for
+  ///    the mixer.
+  void pushHandleLayers(LayerLink? startHandle, LayerLink? endHandle);
+
+  /// Gets the selected content from this selectable.
+  ///
+  /// Return `null` if nothing is selected.
+  Object? getSelectedContent();
+
+  /// handles the [SelectionEvent] sent to this object.
+  ///
+  /// The subclasses need to update its selection or delegate the selection
+  /// event to its subtree.
+  ///
+  /// The [event] is a subclass of [SelectionEvent]. Use runtime type check to
+  /// determine what kind of event is dispatched to this handler and handle it
+  /// accordingly.
+  SelectionResult dispatchSelectionEvent(SelectionEvent event);
+}
+
+/// An object that can be selected by the [SelectionArea] widget.
+///
+/// This object receives selection events and is responsible to draw selection
+/// highlight. In order to receive the selection event, the subclasses need to
+/// register themselves to [SelectionRegistrar]s. Use the
+/// [SelectionContainer.maybeOf] to get the the selection registrar, and mix
+/// the [SelectionRegisteree] to subscribe to the [SelectionRegistrar]
+/// automatically.
+///
+/// The [Selectable] also paints [LayerLink]s of selection handles in a
```

Comment:
My understanding is that the Selectable doesn't do this and instead whoever mixes this class in has to implement that feature?